massa-models/src/denunciation.rs

```diff
@@ -714,6 +731,203 @@ impl Deserializer<Denunciation> for DenunciationDeserializer {
 
 // End Ser / Der
 
+// Denunciation Id
+
+/// Endorsement ID size in bytes
+pub const DENUNCIATION_ID_SIZE_BYTES: usize = massa_hash::HASH_SIZE_BYTES;
+
+/// endorsement id
+#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct DenunciationId(Hash);
+
+const DENUNCIATION_ID_PREFIX: char = 'D';
+const DENUNCIATION_ID_VERSION: u64 = 0;
+
+impl PreHashed for DenunciationId {}
+
+impl Id for DenunciationId {
+    fn new(hash: Hash) -> Self {
+        Self(hash)
+    }
+
+    fn get_hash(&self) -> &Hash {
+        &self.0
+    }
+}
+
+impl std::fmt::Display for DenunciationId {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let u64_serializer = U64VarIntSerializer::new();
+        // might want to allocate the vector with capacity in order to avoid re-allocation
+        let mut bytes: Vec<u8> = Vec::new();
+        u64_serializer
+            .serialize(&DENUNCIATION_ID_VERSION, &mut bytes)
+            .map_err(|_| std::fmt::Error)?;
+        bytes.extend(self.0.to_bytes());
+        write!(
+            f,
+            "{}{}",
+            DENUNCIATION_ID_PREFIX,
+            bs58::encode(bytes).with_check().into_string()
+        )
+    }
+}
+
+impl std::fmt::Debug for DenunciationId {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
+impl DenunciationId {
+    /// denunciation id to bytes
+    pub fn to_bytes(&self) -> &[u8; DENUNCIATION_ID_SIZE_BYTES] {
+        self.0.to_bytes()
+    }
+
+    /// denunciation id into bytes
+    pub fn into_bytes(self) -> [u8; DENUNCIATION_ID_SIZE_BYTES] {
+        self.0.into_bytes()
+    }
+
+    /// denunciation id from bytes
+    pub fn from_bytes(data: &[u8; DENUNCIATION_ID_SIZE_BYTES]) -> DenunciationId {
+        DenunciationId(Hash::from_bytes(data))
+    }
+}
+
+impl From<&Denunciation> for DenunciationId {
+    fn from(value: &Denunciation) -> Self {
+        match value {
+            Denunciation::Endorsement(endo_de) => {
+                let mut to_hash = Vec::with_capacity(
+                    std::mem::size_of::<u32>() + PUBLIC_KEY_SIZE_BYTES + SLOT_KEY_SIZE,
+                );
+                to_hash.extend(endo_de.public_key.to_bytes());
+                to_hash.extend(endo_de.slot.to_bytes_key());
+                to_hash.extend(endo_de.index.to_le_bytes());
+                DenunciationId(Hash::compute_from(&to_hash))
+            }
+            Denunciation::BlockHeader(blkh_de) => {
+                let mut to_hash = Vec::new();
+                to_hash.extend(blkh_de.public_key.to_bytes());
+                to_hash.extend(blkh_de.slot.to_bytes_key());
+                DenunciationId(Hash::compute_from(&to_hash))
+            }
+        }
+    }
+}
+
+// End Denunciation Id
+
+// Denunciation interest
+
+/// DenunciationInterest variant for endorsement
+#[allow(missing_docs)]
```

Comment:
we should never use allow missing_docs nor fake empty docs to avoid warnings => it guarantees we will forget to document this :)

Reply:
fixed